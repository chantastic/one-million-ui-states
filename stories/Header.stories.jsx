import { Header } from "./Header";

export default {
  title: "Example/Header",
  component: Header,
  parameters: {
    layout: "fullscreen",
  },
};

export const LoggedOut = {};

export const LoggedIn = {
  args: {
    user: {
      name: "Jane Doe",
    },
  },
};

// TODO: 4 Because we're documenting isolated components, it's trivial to set up extreme cases
// export const LoggedInWithLongName = {
//   args: {
//     user: {
//       name: "John Jacob Jingleheimer Schmidt",
//     },
//   },
// };

// TODO: 5 Using paramaters (in addition to args), we can set up full UI cases.
//         Like the combination of a long name in small viewport.
// export const LoggedInWithLongNameOnMobile = {
//   args: LoggedInWithLongName.args,
//   parameters: {
//     viewport: {
//       defaultViewport: "mobile1",
//     },
//   },
// };
