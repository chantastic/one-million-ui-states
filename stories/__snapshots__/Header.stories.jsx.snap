// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Example/Header LoggedIn smoke-test 1`] = `
Object {
  "checked": undefined,
  "children": Array [
    Object {
      "checked": undefined,
      "children": undefined,
      "level": 1,
      "name": "Acme",
      "pressed": undefined,
      "role": "heading",
      "value": undefined,
    },
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "Welcome,",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "Jane Doe",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "!",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "Log out",
      "pressed": undefined,
      "role": "button",
      "value": undefined,
    },
  ],
  "name": "example-header--logged-in",
  "pressed": undefined,
  "role": "WebArea",
  "value": undefined,
}
`;
